app/scripts/controllers/permissions/index.js

```diff
@@ -188,33 +214,54 @@ export class PermissionsController {
    */
   async legacyExposeAccounts (origin, accounts) {
 
-    const permissions = {
-      eth_accounts: {},
+    // accounts are validated by finalizePermissionsRequest
+    if (!origin || typeof origin !== 'string') {
```

Comment:
We might be able to restate this to better express the desired input:

```suggestion
    if (typeof origin !== 'string' || !origin.length) {
```

`typeof` should be able to work with a missing argument and this might more closely match the error message we're providing below.